components/movementsensor/adxl345/adxl345.go

```diff
@@ -294,7 +458,13 @@ func (adxl *adxl345) Accuracy(ctx context.Context, extra map[string]interface{})
 func (adxl *adxl345) Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error) {
 	adxl.mu.Lock()
 	defer adxl.mu.Unlock()
-	return map[string]interface{}{"linear_acceleration": adxl.linearAcceleration}, adxl.err.Get()
+
+	readings := make(map[string]interface{})
+	readings["linear_acceleration"] = adxl.linearAcceleration
```

Comment:
[nit] we don't need linear acceleration in the driver reading function anymore, the higher-up `movementsensor.go` handles that.  

So change 463 to:
```suggestion
	readings, err := movementsensor.Readings(ctx, adxl, extra)
	if err != nil {
		return nil, err
	}
```

and keep the rest.